gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/GraphTraversal.java

```diff
@@ -2829,6 +2830,19 @@ public default <S2> GraphTraversal<S, E> any(final P<S2> predicate) {
         return this.asAdmin().addStep(new AnyStep<>(this.asAdmin(), predicate));
     }
 
+    /**
+     * Filters <code>E</code> lists given the provided {@code predicate}.
+     *
+     * @param predicate the filter to apply
+     * @return the traversal with an appended {@link ListNoneStep}
+     * @see <a href="http://tinkerpop.apache.org/docs/${project.version}/reference/#none-step" target="_blank">Reference Documentation - None Step</a>
+     * @since 3.7.1
+     */
+    public default <S2> GraphTraversal<S, E> none(final P<S2> predicate) {
+        this.asAdmin().getBytecode().addStep(Symbols.listNone, predicate);
+        return this.asAdmin().addStep(new ListNoneStep<>(this.asAdmin(), predicate));
```

Comment:
I realize `NoneStep` is already taken and has special meaning that is not easily removed at this moment. Given that `none()` is just `not(any())` and the code for this `ListNoneStep` and `AnyStep`, i think it would be better to just give `AnyStep` a negation configuration. Then you get rid of `ListNoneStep`.